### PR TITLE
Improve message about insecure S3 settings

### PR DESCRIPTION
**Pull request by DaveCTurner**
Clarifies that insecure settings are stored in plaintext and must not be
used. Also removes the mention of the (wrong) system property from the
error message if insecure settings are not permitted.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/settings/SecureSetting.java [hunk 1]
```diff
@@ -185,9 +185,7 @@ private InsecureStringSetting(String name) {
         @Override
         public SecureString get(Settings settings) {
             if (ALLOW_INSECURE_SETTINGS == false && exists(settings)) {
-                throw new IllegalArgumentException(
-                    "Setting [" + name + "] is insecure, " + "but property [allow_insecure_settings] is not set"
-                );
+                throw new IllegalArgumentException("Setting [" + name + "] is insecure and no longer supported");
```
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/common/settings/SecureSetting.java:188]:
```suggestion
                throw new IllegalArgumentException("Setting [" + name + "] is insecure, use the elasticsearch keystore instead");
```

**Review by henningandersen — APPROVED** [review]
LGTM.

File: server/src/main/java/org/elasticsearch/common/settings/SecureSetting.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/common/settings/SecureSetting.java]: :+1: good point

**Comment by DaveCTurner** [comment]
Backports:
- #116954 (8.17)
- #116955 (8.16)
- #116956 (7.17)

**Pull request merged**